### ISSUE TITLE
Training loss is mixed with valid loss

### ISSUE THREAD

**Issue by freewym** [pytorch/fairseq#1632]
## 🐛 Bug

After the latest introduction of metrics, the training loss printed as log is different from before, which I suspect is because  its value is somewhat mixed with valid loss

### To Reproduce

Steps to reproduce the behavior (**always include the command you ran**):

You can reproduce it by observing the difference of training losses before/after setting `--disable-validation`

### Environment

 - fairseq Version (e.g., 1.0 or master): master
 - PyTorch Version (e.g., 1.0): 1.1
 - OS (e.g., Linux):
 - How you installed fairseq (`pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

**Comment by myleott** [comment]
Good catch, this was actually introduced in 60fbf6, with the switching of lines 166/167. Will merge a fix soon.
